src/mixins/java/org/spongepowered/common/mixin/core/world/level/chunk/LevelChunkMixin.java

```diff
@@ -352,4 +352,14 @@ public String toString() {
         return this.impl$cacheKey;
     }
 
+    @Override
+    public boolean bridge$spawnEntity(final org.spongepowered.api.entity.Entity entity) {
+        final net.minecraft.world.entity.Entity mcEntity = (net.minecraft.world.entity.Entity) entity;
+        if (this.chunkPos.equals(new ChunkPos(mcEntity.blockPosition()))) {
```

Comment:
I wonder if it is faster to bitshift the block position and do equals on the ints. I imagine so.